### PR TITLE
New auth providers

### PR DESCRIPTION
**Pull request by haowang1013**
Added:
- discord
- twitter
- facebook
- google
- steam (openid)

### CODE REVIEW

**Comment by judgeaxl** [comment]
There's a significant difference here in how this is implemented compared to other auth methods in that you send in the code to get a token. The code is always single-use, so if this were to be used with the website, then the website cannot use the session too, but has to run the login twice, as the token returned from the authentication is for Drift, not for the logged in account. The other implementations take the final ID or access token, validate it, and use it for login. This also means we cannot use the token a website or client might have already to perform a login, e.g. if we use Google login in the client through the Google OSS, we cannot use that token to log the player in.

**Review by judgeaxl — CHANGES_REQUESTED** [review]
Nothing strictly wrong in the implementation as is, but as I commented elsewhere it's a big shift from how all other providers work, where we get the identity from the caller and validate it.

This makes these APIs behave differently from all the others, and make them incompatible with any client that uses a flow which doesn't return the authorisation code directly, or which needs its own copy of the ID or access token, since the code itself is single-use as per the OAuth spec.

To follow the existing pattern, the client should pass in the ID or access token instead, and the backend should validate it. Otherwise we'll need to add additional APIs for the same providers later.

File: driftbase/auth/__init__.py [hunk 1]
```diff
@@ -14,6 +14,11 @@
     'eos': 'driftbase.auth.eos',
     'ethereum': 'driftbase.auth.ethereum',
     'cognito': 'driftbase.auth.cognito',
+    'discord': 'driftbase.auth.discord',
+    # 'twitter': 'driftbase.auth.twitter',
```
Comment (judgeaxl) [inline driftbase/auth/__init__.py:18]: Please add a note about why these are disabled for the next reader. I'd suggest adding a 400 error or exception on trying to enable these too before they are fixed.

Comment (haowang1013) [reply driftbase/auth/__init__.py:19]: Fixed in the latest

**Review by judgeaxl — APPROVED** [review]
When are the rest going to be needed?

**Comment by haowang1013** [comment]
> When are the rest going to be needed?

Unclear, might need twitter soon

**Pull request merged**